### ISSUE TITLE
JSONService.py does not work with the built-in json module of Python 2.6 and later

### ISSUE THREAD

**Issue by pyjsorg** [pyjs/original-pyjs#272]
A recent change to JSONService.py caused it, when running with pyjd, to import loads, dumps and JSONDecodeError from simplejson, instead of loads, dumps and JSONDecodeException from jsonrpc.json. This does not work with Python 2.6, which has a built-in json module which is derived from a slightly earlier version of simplejson (and is called json, not simplejson). I suggest a modification to try importing dumps and loads from json (Python 2.6 and later), failing that from simplejson and failing that assuming that pyjs is being used. The use of a named exception for decode errors would be dropped because of the complication of deciding which exceptions to catch. In the circumstance, a catch-all exception seems justified.

A patch file for the suggested changes is attached. It has been tested on Python 2.6.4 for both pyjd and pyjs but not with Python 2.5 and simplejson.

_Original issue: http://code.google.com/p/pyjamas/issues/detail?id=447 (July 06, 2010 20:25:44)_

**Comment by pyjsorg** [comment]
_From luke.lei...@gmail.com on July 07, 2010 10:32:21:_
thanks charles.  i whacked in something similar last night, could you possibly git pull and check it?  thx.

**Comment by pyjsorg** [comment]
_From jpcharle...@gmail.com on July 07, 2010 13:56:35:_
Sorry, lkcl, your version doesn't work in pyjd on my system. The problem, I think, is that jsonrpc.json doesn't support the object_hook keywork argument in loads (line 179 of JSONRPCService.py). So, no point in looking at jsonrpc. That leaves json and simplejson. I think one should check for json first because, if not now, then very soon, most users will be using Python 2.6 or 2.7 and so will have json. The exception will only occur for the remaining (ever fewer) Python 2.5 users who need to have installed simplejson.
So, in summary, I see no reason to change the patch I sent.

**Comment by pyjsorg** [comment]
_From luke.lei...@gmail.com on July 08, 2010 15:39:42:_
ack.  good point.  swapped.  removed jsonrpc.json - no point really.

**Comment by pyjsorg** [comment]
_From jpcharle...@gmail.com on July 23, 2010 17:48:04:_
I see you have applied the agreed changes. It works for me. Perhaps this issue can be closed now.

**Comment by pyjsorg** [comment]
_From luke.lei...@gmail.com on September 23, 2010 13:01:28:_
ack.
